addons/html_editor/static/src/utils/dom_info.js

```diff
@@ -489,7 +489,7 @@ export function isAllowedContent(parentBlock, nodes) {
  * @returns {boolean}
  */
 export function isEmptyBlock(blockEl) {
-    if (!blockEl || blockEl.nodeType !== Node.ELEMENT_NODE) {
+    if (!blockEl || blockEl.nodeType !== Node.ELEMENT_NODE || !isBlock(blockEl)) {
```

Comment:
@adch-odoo @dmo-odoo 
`isBlock` already checks for `!node || node.nodeType !== Node.ELEMENT_NODE`, so the first two checks are not really necessary (unless you want to keep this function robust and not rely on `EmptyBlocks` implementation)

Also, we have a test suite for our utils (see `addons/html_editor/static/tests/utils/dom_info.test.js` or `addons/html_editor/static/tests/utils/block.test.js`. I would suggest adding unit tests for this function: nullish argument, non-block argument, icon, image, etc.

I would also suggest adding a test case for the reported problem (inserting icons one after another) to avoid regressions.